### ISSUE TITLE
There is no way to enable the necessary features of the SQL Server

### ISSUE THREAD

**Issue by MirolimMajidov** [microsoft/winget-cli#3522]
### Brief description of your issue

I'm using the WinGet configuration for installing some necessary applications with components by creating a YAML file and using that from the Dev Home. I did all of them except the SQL Server features. I'm able to install the SQL server, but I could not enable (Install) the "Full-text search" and "FileStream" features of the SQL Server. 

### Steps to reproduce

There is my configuration file. The first step is passing but the second is not: 

# yaml-language-server: $schema=https://aka.ms/configuration-dsc-schema/0.2
properties:
  resources:
    - resource: Microsoft.WinGet.DSC/WinGetPackage
      id: sqlServerPackage
      directives:
        description: Install Microsoft SQL Server 2022 Developer
        allowPrerelease: true
      settings:
        id: Microsoft.SQLServer.2022.Developer
        source: winget
    - resource: Microsoft.SqlServerDsc/SqlServerFeature
      dependsOn:
        - sqlServerPackage
      directives:
        description: Enable necessary SQL features
      settings:
        features: 'FULLTEXT'
        filestream: true
  configurationVersion: 0.2.0first

### Expected behavior

Somehow we should be doing that because a lot of times just installing the SQL Server is not enough to use that without enabling the features of changing the configuration. 

At least if there is a way to do that by a PowerShell script, please show me an example of how to call the PowerShell script from the Yaml winget configuration. I did not find any thing for that.

### Actual behavior

It is failing from the second step without any error.

### Environment

```shell
I am using the lest version of WinGet and Windows 11
```

**Comment by stephengillie** [comment]
Feature request: Support configuration files. (We might already have this request in another Issue.)

**Comment by MirolimMajidov** [comment]
> Feature request: Support configuration files. (We might already have this request in another Issue.)

@stephengillie 
Our company has close to 100 software engineers. So, we are planning to use the Dev Home to decrease the setting up the dev environments. As I said, we have prepared the configuration files to install and customize all necessary applications except the MS SQL Server. That means there is still manual work to install the  SQL Server.

You already said you have plans to implement this feature. How long do you think it might take?
If it takes more than a week or a month, could you please show me a workaround until implementing this functionality?

**Comment by stephengillie** [comment]
Hi @MirolimMajidov,

Thanks for explaining your situation and goals. I'm not sure even if this is on our roadmap currently. I'll defer to @denelon for the best info about our future feature plans.

**Comment by mdanish-kh** [comment]
[Policy] Command-Configure

**Comment by denelon** [comment]
@MirolimMajidov, I did a quick search for Microsoft.SqlServerDsc at the PowerShell gallery. I didn't see any results. Where did you locate that PowerShell module?

**Comment by denelon** [comment]
There is a [DSC Script Resource](https://learn.microsoft.com/powershell/dsc/reference/resources/windows/scriptresource?view=dsc-1.1). The primary thing to be cautious of is to make sure your script is idempotent so it can be run multiple times without causing a failure or other unintended side effects.

**Comment by MirolimMajidov** [comment]
@denelon I'm not familiar with the Microsoft.SqlServerDsc, but it seems we can do that with DSC SqlServer script.

**Comment by denelon** [comment]
@johlju I'm not familiar with SQL DSC. Do you have any docs or anything we could point a user at?

We could also work on a sample configuration over at [Dev Home Sample Configurations](https://aka.ms/dsc.yaml).

**Comment by johlju** [comment]
I have no knowledge of how the configuration file mentioned above is used and where the source for `Microsoft.SqlServerDsc/SqlServerFeature` is comig from. Please provide more details. But I know it is not possible to enable features post-install, other than running install again with tre correct arguments. The manifest  [Microsoft.SQLServer.2022.Developer.installer.yaml](https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/SQLServer/2022/Developer/16.0.1000.6/Microsoft.SQLServer.2022.Developer.installer.yaml) does not seem to have any argument options at all? A manifest needs to handle all of these properties in a real scenario: https://learn.microsoft.com/en-us/sql/database-engine/install-windows/install-sql-server-from-the-command-prompt

There is DSC resource SqlSetup that can install SQL Server: https://github.com/dsccommunity/SqlServerDsc/wiki/SqlSetup

There are a command (`Install-SqlDscServer`) in [SqlServerDsc](https://www.powershellgallery.com/packages/SqlServerDsc/16.5.0) that can install SQL Server using PowerShell (see comment-based help for docs, there are no community docs as of yet). The source: https://github.com/dsccommunity/SqlServerDsc/blob/main/source/Public/Install-SqlDscServer.ps1

Not sure if this helps anything.

**Comment by johlju** [comment]
There is also a new DSC resource in the works that better work with the latests SQL Server (support more options): https://github.com/dsccommunity/SqlServerDsc/pull/1912

**Comment by tristanbarcelon** [comment]
> [@MirolimMajidov](https://github.com/MirolimMajidov), I did a quick search for Microsoft.SqlServerDsc at the PowerShell gallery. I didn't see any results. Where did you locate that PowerShell module?

This is the one I use: https://www.powershellgallery.com/packages/SqlServerDsc

**Comment by denelon** [comment]
Thanks for the detailed configuration and repro. This isn't really a WinGet-specific request, so we're going to close it here — but there are concrete paths to unblock the scenario.

**Issue summary**
- After installing SQL Server via `Microsoft.WinGet.DSC/WinGetPackage`, enabling features like Full-Text Search and FileStream via `Microsoft.SqlServerDsc/SqlServerFeature` fails silently. The ask is a way to enable SQL Server features from a WinGet Configuration file.

**Why this is out of scope for WinGet**
- WinGet Configuration is a **host for DSC resources** — it invokes whatever resources you reference and reports their results. Enabling SQL Server features is the responsibility of a DSC resource that knows how to configure SQL Server, not of WinGet itself. The failure here is in the resource layer, not in WinGet's configuration processing.

**Recommended paths**
1. **Use the appropriate SQL Server DSC resource.** Feature enablement (SQLENGINE, FULLTEXT, FileStream, etc.) is handled by the SqlServerDsc module's setup resource (e.g. `SqlSetup`) rather than a standalone "feature" resource. Behavior, bugs, and enhancement requests for that belong in that module's repo. If a suitable command-based DSC v3 resource for this scenario doesn't exist, authoring one (or filing for it) in the relevant DSC resource project is the right fix.
2. **Interim unblock with a Script resource.** Until a resource cleanly covers your case, you can run the SQL Server setup with the desired feature flags from configuration using the `Script` resource (from `PSDscResources`), which lets you provide `GetScript` / `TestScript` / `SetScript`. Sketch:

```yaml
    - resource: PSDscResources/Script
      dependsOn:
        - sqlServerPackage
      directives:
        description: Enable SQL Server features
      settings:
        GetScript: "@{ Result = '' }"
        TestScript: "return $false"
        SetScript: |
          # invoke SQL Server setup with the desired features, e.g.:
          & 'C:\path\to\setup.exe' /q /Action=Install /Features=FULLTEXT /FILESTREAMLEVEL=1 /IAcceptSQLServerLicenseTerms
```

Since the request is about SQL Server feature configuration via a DSC resource rather than WinGet behavior, closing as not planned.

**Related**
- microsoft/winget-dsc (for authoring/adopting DSC resources)

*Drafted with the help of GitHub Copilot.*
